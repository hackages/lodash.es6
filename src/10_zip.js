/**
 * Creates an array of grouped elements, the first of which contains the first elements of the given arrays, the second
 * of which contains the second elements of the given arrays, and so on.
 */
export function zip(...arrays) {}
