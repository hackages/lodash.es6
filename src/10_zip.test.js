import {zip} from './10_zip'

describe('zip', () => {
  it('should zip', () => {
    expect(zip(['a', 'b'], [1, 2], [true, false])).toMatchSnapshot()
    expect(zip(['a'], [1, 2], [true, true, false])).toMatchSnapshot()
    expect(zip(['a', 'b', 'c'], [1, 2], [true])).toMatchSnapshot()
    expect(zip(['hello'], [undefined])).toMatchSnapshot()
    expect(zip([32], 'ohoh')).toMatchSnapshot()
    expect(zip([32])).toMatchSnapshot()
    expect(zip(32)).toMatchSnapshot()
  })
})
