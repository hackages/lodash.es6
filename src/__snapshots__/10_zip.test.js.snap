// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zip should zip 1`] = `
Array [
  Array [
    "a",
    1,
    true,
  ],
  Array [
    "b",
    2,
    false,
  ],
]
`;

exports[`zip should zip 2`] = `
Array [
  Array [
    "a",
    1,
    true,
  ],
  Array [
    undefined,
    2,
    true,
  ],
  Array [
    undefined,
    undefined,
    false,
  ],
]
`;

exports[`zip should zip 3`] = `
Array [
  Array [
    "a",
    1,
    true,
  ],
  Array [
    "b",
    2,
    undefined,
  ],
  Array [
    "c",
    undefined,
    undefined,
  ],
]
`;

exports[`zip should zip 4`] = `
Array [
  Array [
    "hello",
    undefined,
  ],
]
`;

exports[`zip should zip 5`] = `
Array [
  Array [
    32,
  ],
]
`;

exports[`zip should zip 6`] = `
Array [
  Array [
    32,
  ],
]
`;

exports[`zip should zip 7`] = `Array []`;
